### PR TITLE
rust 1.25.0

### PR DESCRIPTION
**Pull request by alex**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/rust.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@ class Rust < Formula
   revision 1
```
Comment (commitay) [inline Formula/rust.rb]: The revision needs to be removed

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
```
Error: Failed changing dylib ID of /usr/local/Cellar/rust/1.25.0/share/rust/rust_src/llvm-emscripten/test/Object/Inputs/macho-invalid-dylib-id-more-than-one
  from foo
    to /usr/local/opt/rust/share/rust/rust_src/llvm-emscripten/test/Object/Inputs/foo
Error: Failed to fix install linkage
```

So apparently as part of rust sources, we're now shipping some object files and binaries… and even some invalid ones 😱

**Comment by fxcoudert** [comment]
It even makes our audit fail:

```
        Error: Failed to read Mach-O binary: /usr/local/Cellar/rust/1.25.0/share/rust/rust_src/llvm-emscripten/test/Object/Inputs/macho-invalid-dyld-name_offset-toobig
Error: undefined method `partition' for nil:NilClass
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/load_commands.rb:284:in `initialize'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/load_commands.rb:530:in `new'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/load_commands.rb:530:in `initialize'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/load_commands.rb:198:in `new'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/load_commands.rb:198:in `new_from_bin'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/macho_file.rb:496:in `block in populate_load_commands'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/macho_file.rb:485:in `times'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/macho_file.rb:485:in `populate_load_commands'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/macho_file.rb:235:in `populate_fields'
/usr/local/Homebrew/Library/Homebrew/vendor/macho/macho/macho_file.rb:46:in `initialize'
```

**Comment by ilovezfs** [comment]
The
```
 rm_rf Dir["src/{llvm,test,librustdoc,etc/snapshot.pyc}"]
```
is supposed to handle that. So probably just need to throw some new dirs at it.

**Comment by ilovezfs** [comment]
CC @woodruffw that audit fail is nifty

**Comment by woodruffw** [comment]
Yeah, my understanding is that Rust ships with all of binaries that LLVM uses in its unit tests. Updating the glob should fix the problem.

Looks like the audit failure begins when we run `check_openssl_links`, which tries to iterate over all the Mach-O files and then explodes when it gets to a malformed one. I'll take this malformed binary and see if I can make the inner error thrown by ruby-macho better, but at least this confirms that `mach_data`'s `rescue` works correctly :slightly_smiling_face: :

```ruby
    rescue MachO::NotAMachOError
      # Silently ignore errors that indicate the file is not a Mach-O binary ...
      []
    rescue
      # ... but complain about other (parse) errors for further investigation.
      if ARGV.homebrew_developer?
        onoe "Failed to read Mach-O binary: #{self}"
        raise
      end
      []
    end
```

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Pull request closed without merging**